.github/workflows/coverage.yml

```diff
@@ -0,0 +1,45 @@
+name: Coverage
+on: [push]
```

Comment:
I think we should only run this on push to `main`. What do you think? https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#using-filters